src/dlgProfilePreferences.cpp

```diff
@@ -3380,3 +3416,140 @@ void dlgProfilePreferences::slot_changeGuiLanguage(const QString &language)
 
     label_languageChangeWarning->show();
 }
+
+void dlgProfilePreferences::slot_changePlayerRoomStyle(const int index)
+{
+    Host* pHost = mpHost;
+    if (!pHost || !mpHost->mpMap) {
+        return;
+    }
+
+    int style = index;
+    switch (index) {
+    case 1: // Red ring
+        pushButton_playerRoomPrimaryColor->setEnabled(false);
+        pushButton_playerRoomSecondaryColor->setEnabled(false);
+        spinBox_playerRoomInnerDiameter->setEnabled(true);
+        break;
+
+    case 2: // Blue-yellow ring
+        pushButton_playerRoomPrimaryColor->setEnabled(false);
+        pushButton_playerRoomSecondaryColor->setEnabled(false);
+        spinBox_playerRoomInnerDiameter->setEnabled(true);
+        break;
+
+    case 3: // Custom ring
+        pushButton_playerRoomPrimaryColor->setEnabled(true);
+        pushButton_playerRoomSecondaryColor->setEnabled(true);
+        spinBox_playerRoomInnerDiameter->setEnabled(true);
+        break;
+
+    default:
+        style = 0;
+        [[clang::fallthrough]];
+    case 0: // "Original"
+        pushButton_playerRoomPrimaryColor->setEnabled(false);
+        pushButton_playerRoomSecondaryColor->setEnabled(false);
+        spinBox_playerRoomInnerDiameter->setEnabled(false);
+    }
+    setButtonColor(pushButton_playerRoomPrimaryColor, pHost->mpMap->mPlayerRoomColorPrimary);
+    setButtonColor(pushButton_playerRoomSecondaryColor, pHost->mpMap->mPlayerRoomColorSecondary);
+    pHost->mPlayerRoomColorPrimary = pHost->mpMap->mPlayerRoomColorPrimary;
+    pHost->mPlayerRoomColorSecondary = pHost->mpMap->mPlayerRoomColorSecondary;
+    pHost->mpMap->mPlayerRoomStyle = static_cast<quint8>(style);
+    pHost->mPlayerRoomStyle = static_cast<quint8>(style);
+    if (!mpHost->mpMap->mpMapper || !mpHost->mpMap->mpMapper->mp2dMap) {
+        return;
+    }
+    pHost->mpMap->mpMapper->mp2dMap->setPlayerRoomStyle(style);
+    // And update the displayed map:
+    mpHost->mpMap->mpMapper->mp2dMap->update();
+}
+
+void dlgProfilePreferences::slot_setPlayerRoomPrimaryColor()
+{
+    Host* pHost = mpHost;
+    if (!pHost || !mpHost->mpMap || !mpHost->mpMap->mpMapper || !mpHost->mpMap->mpMapper->mp2dMap) {
+        return;
+    }
+
+    setPlayerRoomColor(pushButton_playerRoomPrimaryColor, mpHost->mpMap->mPlayerRoomColorPrimary);
+    if (comboBox_playerRoomStyle->currentIndex() != 3) {
+        return;
+    }
+
+    // The current setting IS for the custom color - so use it straight away:
+    mpHost->mpMap->mpMapper->mp2dMap->setPlayerRoomStyle(3);
+    // And update the displayed map:
+    mpHost->mpMap->mpMapper->mp2dMap->update();
+}
+
+void dlgProfilePreferences::slot_setPlayerRoomSecondaryColor()
+{
+    Host* pHost = mpHost;
+    if (!pHost || !mpHost->mpMap || !mpHost->mpMap->mpMapper || !mpHost->mpMap->mpMapper->mp2dMap) {
```

Comment:
This setting should be adjustable even when the map isn't open, right?

Reply:
Debatable, I think. After all why show the controls if you are Mudding without a map - I know I wouldn't but not everyone thinks like that...

Reply:
Well - because you haven't yet opened the map?

Reply:
Ah, looking at how the code is structured that slot directly alters the 2D mapper by calling a method in the `T2DMap` class - which can't happen if there is no map loaded. Going to have to rethink this...